### PR TITLE
Fix: freeze Jinja2 version

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Jinja2 just released a new version and caused an internal error in the docs build. Pinning the dependency to fix it.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: requirements/docs.txt [hunk 1]
```diff
@@ -12,5 +12,6 @@ sphinx-paramlinks>=0.5.1
 sphinx-togglebutton>=0.2
 sphinx-copybutton>=0.3
 typing-extensions  # already in `requirements.txt` but the docs CI job does not install it
+jinja2<3.1.0
```
Comment (rohitgr7) [inline requirements/docs.txt:15]: should we specify any minimum version?

**Review by rohitgr7 — APPROVED** [review]

File: requirements/docs.txt (hunk 1 above)
Comment (ethanwharris) [reply requirements/docs.txt:15]: Could set minimum to 3.0.0 :smiley:

Comment (ethanwharris) [reply requirements/docs.txt]: Done!

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
Is it only for PL or all other projects?

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]

**Comment by ethanwharris** [comment]
@Borda seems to be affecting everything. Same issue on Flash, I guess will hit TM too

**Comment by Borda** [comment]
> @Borda seems to be affecting everything. Same issue on Flash, I guess will hit TM too

well thinking about really freezing all our docs requirements across projects as it is a specific task I do not expect that regular user would need building docs and if all these requirements are rather light... :rabbit:

**Pull request merged**